solr/solrj/src/test/org/apache/solr/client/solrj/impl/HttpSolrJdkClientTest.java

```diff
@@ -278,6 +281,15 @@ public void testUpdateDefault() throws Exception {
 
   @Test
   public void testUpdateXml() throws Exception {
+    testUpdateXml(false);
+  }
+
+  @Test
+  public void testUpdateXmlWithHttp11() throws Exception {
+    testUpdateXml(true);
+  }
```

Comment:
Could we parameterize the test instead so that the whole test runs in both modes?

Reply:
Or just use randomization -- easy and adequate even if a bit annoying to debug a failure.

Reply:
I just read up a little on Junit4 Parameterized tests, and well, it seems so much more complicated than what I have done here.  Yes, randomization will make the test suite run some ms faster, but I'd rather better know that nothing broke with 1 green test run.  In my opinion while randomization is very useful we perhaps overuse it sometimes.